src/display/display_utils.js

```diff
@@ -614,6 +614,45 @@ function getCurrentTransformInverse(ctx) {
   return [a, b, c, d, e, f];
 }
 
+/**
+ * @param {HTMLDivElement} div
+ * @param {PageViewport} viewport
+ * @param {boolean} mustFlip
+ * @param {boolean} mustRotate
+ */
+function setLayerDimensions(
+  div,
+  viewport,
+  mustFlip = false,
+  mustRotate = true
+) {
+  if (!viewport.viewBox) {
+    if (mustRotate) {
+      div.setAttribute("data-main-rotation", viewport.rotation);
+    }
+    return;
+  }
+
+  const [pageLLx, pageLLy, pageURx, pageURy] = viewport.viewBox;
+  const pageWidth = pageURx - pageLLx;
+  const pageHeight = pageURy - pageLLy;
+  const { style } = div;
+  const widthStr = `calc(var(--scale-factor) * ${pageWidth}px)`;
+  const heightStr = `calc(var(--scale-factor) * ${pageHeight}px)`;
+
+  if (!mustFlip || viewport.rotation % 180 === 0) {
+    style.width = widthStr;
+    style.height = heightStr;
+  } else {
+    style.width = heightStr;
+    style.height = widthStr;
+  }
```

Comment:
Can we perhaps avoid duplicating the rotation-handling with something like the below, which should be a tiny bit less code overall?

```suggestion
  if (viewport.viewBox) {
    const [pageLLx, pageLLy, pageURx, pageURy] = viewport.viewBox;
    const pageWidth = pageURx - pageLLx;
    const pageHeight = pageURy - pageLLy;
    const { style } = div;
    const widthStr = `calc(var(--scale-factor) * ${pageWidth}px)`;
    const heightStr = `calc(var(--scale-factor) * ${pageHeight}px)`;

    if (!mustFlip || viewport.rotation % 180 === 0) {
      style.width = widthStr;
      style.height = heightStr;
    } else {
      style.width = heightStr;
      style.height = widthStr;
    }
  }
```